### ISSUE TITLE
BSS segment not cleared on firmware load

### ISSUE THREAD

**Issue by dinuxbg** [dinuxbg/gnupru#46]
BSS segment is not cleared by the Linux kernel firmware loader.

Host: Linux BeagleBone 5.10.120-ti-arm64-r63 #1bullseye SMP Fri Sep 2 01:18:17 UTC 2022 aarch64 GNU/Linux

**Comment by dinuxbg** [comment]
Also present in latest BB-AI64 kernel:
```
 Linux BeagleBone 5.10.153-ti-arm64-r86 #1bullseye SMP Wed Jan 4 19:47:11 UTC 2023 aarch64 GNU/Linux
 ```

**Comment by dinuxbg** [comment]
Seems intentional behavior. From https://github.com/beagleboard/linux/blob/5.10.153-ti-arm64-r86/drivers/remoteproc/pru_rproc.c#L974 :

> /* skip the memzero logic performed by remoteproc ELF loader */

This post has some background: https://lore.kernel.org/lkml/1b661532-6846-1c60-6dc7-11bd23709e20@ti.com/

**Comment by dinuxbg** [comment]
Sent a fix to upstream for review: https://sourceware.org/pipermail/binutils/2023-January/125836.html

**Comment by dinuxbg** [comment]
Fixed with release 2023.01
